content/blog/2020-12-tokio-1-0.md

```diff
@@ -0,0 +1,150 @@
+For example, [Discord] used Tokio to reduce tail latencies 5x. [Fly.io] found
+that, with Tokio, they can effortlessly meet their performance requirements and
+focus on delivering features to their customers. For [ZCash], building on Tokio
+enables them to design misuse-resistant APIs. For AWS, their [Lambda][lambda]
+team used Tokio to implement a more reliable and flexible version of
+their stateful router.
+
+[Discord]: https://blog.discord.com/why-discord-is-switching-from-go-to-rust-a190bbca2b1f
+[Fly.io]: https://fly.io
+[ZCash]: https://www.zfnd.org/blog/futures-batch-verification/
+[lambda]: https://aws.amazon.com/lambda/
+
+The [tutorial] is the best place to learn how to get started with Tokio.
+
+[tutorial]: https://tokio.rs/tokio/tutorial
+
+# A stability guarantee
+
+We first [announced] Tokio over four years ago. Since then, it has evolved
+significantly. The most notable change happened a year ago with the addition of
+[async and await][async_await] to Rust. Today, Tokio is easier to use and more
+powerful. This evolution has also caused some friction. It has required
+libraries to track these changes. It could also result in confusing error
+messages when accidentally depending on multiple versions of Tokio.
+
+[announced]: https://medium.com/@carllerche/announcing-tokio-df6bb4ddb34
+[async_await]: https://tokio.rs/blog/2019-11-tokio-0-2
+
+The Tokio 1.0 release ends this churn. As part of the release, we are committing
+to providing a stable foundation for the ecosystem. We have no current plans for
+Tokio 2.0 and we commit to holding back on a Tokio 2.0 release for at least 3
+years. We also commit to maintain the Tokio 1.0 branch for a minimum of 5 years.
+Tokio will keep a rolling MSRV (minimum supported rust version) policy of 6
+months. When increasing the MSRV, the new Rust version must have been released
+at least six months ago.
+
+This stability does not mean that Tokio will stagnate. We still have a lot of
+work to do.
+
+# 2021
+
+We expect to focus on a few areas this coming year: `Stream`, `io_uring`, tracing
+integration, and the Tokio stack.
+
+## `Stream`
+
+Today, the [`tokio-stream`] crate provides asynchronous iteration utilities
+based on the Stream trait. An [RFC] to move the [`Stream` trait][Stream] from
+[`futures-core`] to the Rust standard library is pending approval.
+
+Once the standard library provides the Stream trait, Tokio’s stream utilities
+will move into the tokio crate itself.
+
+[`tokio-stream`]: https://docs.rs/tokio-stream
+[RFC]: https://github.com/rust-lang/rfcs/pull/2996
+[Stream]: https://docs.rs/futures-core/0.3/futures_core/stream/trait.Stream.html
+[`futures-core`]: https://docs.rs/futures-core/
+
+## `io_uring`
+
+[`io_uring`] is a new Linux interface providing asynchronous operations for all I/O,
+including disk, while reducing required syscalls. Currently, on Linux, Tokio
+uses the epoll interface, which notoriously does not work for disk access. To
```

Comment:
A link explaining this in more detail would be good. I keep forgetting what is and isn't possible w/ epoll which should be evident by my questions about it on Discord!

Reply:
The linked PDF should cover the details. It includes a discussion of existing ways to do I/O on linux.